### PR TITLE
Add rule react/jsx-curly-spacing

### PR DESCRIPTION
**Pull request by fellax1**
### Problem
We have the rule that curly braces e.g. describing objects should be padded with a space, but the rule is not enforced in jsx.

### Solution
Add the rule for jsx.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const moduleConfig = {
     node: true,
     ...(hasES2022Support ? { es2022: true } : { es6: true }),
   },
-  plugins: [ "eslint-plugin-n", "import" ],
+  plugins: [ "eslint-plugin-n", "import", "eslint-plugin-react" ],
```
Comment (markusn) [inline index.js:22]: We have the react.js file that adds the plugin already (by extending @bonniernews/eslint-config/react)

**Review by markusn — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (fellax1) [reply index.js:22]: I have an issue with the tests failing if I don't add it here. Can I add it somewhere else?

Comment (markusn) [reply index.js:22]: "react/jsx-curly-spacing": [ "error", { when: "never", children: { when: "always" } } ],  is in the wrong file, it should be in react-rules.js

Comment (fellax1) [reply index.js:22]: Ah, thanks!

**Review by linus — APPROVED** [review]
{{ ⭐ 👑 💯 }}

**Review by markusn — APPROVED** [review]

**Pull request merged**